src/Raven.Server/Documents/Queries/Graph/SingleEdgeMatcher.cs

```diff
@@ -69,6 +65,40 @@ public void SingleMatch(Match left, string alias)
             }
         }
 
+        private void EdgeMatch(Match match, BlittableJsonReaderObject blittableJsonReaderObject)
+        {
+            if(Edge.Where == null)
+                return;
+            if (Edge.Where is MethodExpression method)
+            {
+                if (EdgeMatchExact(blittableJsonReaderObject, method))
+                {
+                    AddEdgeAfterFiltering(match, blittableJsonReaderObject, Edge.Project.FieldValue);
+                }
+                return;
+            }
+            if (Edge.Where.IsMatchedBy(blittableJsonReaderObject, QueryParameters) != false)
+            {
+                AddEdgeAfterFiltering(match, blittableJsonReaderObject, Edge.Project.FieldValue);
+            }
+        }
+
+        private bool EdgeMatchExact(BlittableJsonReaderObject blittableJsonReaderObject, MethodExpression methodExpression)
+        {
+            if (string.Equals(methodExpression.Name.Value, "exact", StringComparison.OrdinalIgnoreCase) == false)
+                return false;
```

Comment:
You want me to throw? I was asked not to throw from queries before.

Reply:
Throwing when the query itself is invalid - okay.
In this case, you are using a method that isn't supported.

Throwing when the query is valid and some of the docs do not match the expected shape - not okay.

It is about consistency.
If we have a query that would fail for everything, great. Users won't issue that.
IF we have a query that can fail if the data is bad, users might run into that in production, which is _bad_